Prog/NCHousing_needs_units_targets.sas

```diff
@@ -0,0 +1,927 @@
+
+		  set NCarea_&year.
+		    (keep= rent rentgrs pernum gq ownershpd county2_char
+		     where=(pernum=1 and gq in (1,2) and ownershpd in ( 22 )));
+		     
+		  Ratio_rentgrs_rent_&year. = rentgrs / rent;
+		 
+		run;
+
+		proc means data=Ratio_&year.;
+		  var  Ratio_rentgrs_rent_&year. rentgrs rent;
+		  output out=Ratio_&year (keep=Ratio_rentgrs_rent_&year.) mean=;
+		run;
+
+
+data Housing_needs_baseline_&year._1;
+
+  set NCarea_&year.
+        (keep=year serial pernum MET2013 hhwt hhincome numprec UNITSSTR BUILTYR2 bedrooms gq ownershp owncost ownershpd rentgrs valueh county2_char afact afact2
+         where=(pernum=1 and gq in (1,2) and ownershpd in ( 12,13,21,22 )));
+
+		 *adjust all incomes to 2017 $ to match use of 2017 family of 4 income limit in projections (originally based on use of most recent 5-year IPUMS; 
+	MID= put(MET2013, 5.);
+
+	if MET2013= 0 then do;
+	MID="99999";
+	end;
+
+run;
+
+proc sort data= Housing_needs_baseline_&year._1;
+by MID;
+run;
+
+data Housing_needs_baseline_&year._2;
+merge Housing_needs_baseline_&year._1(in=a) NC17_limits;
+if a;
+by MID;
+run;
+
+data Housing_needs_baseline_&year.;
+set Housing_needs_baseline_&year._2;
+	  if hhincome ~=.n or hhincome ~=9999999 then do; 
+		 %dollar_convert( hhincome, hhincome_a, &year., 2017, series=CUUR0000SA0 )
+	   end; 
+
+	*create HUD_inc - uses 2017 limits but has categories for 120-200% and 200%+ AMI; 
+
+  %Hud_inc_NCState( hhinc=hhincome_a, hhsize=numprec )  
+
+run; 
+
+data Housing_needs_baseline_&year._3;
+  set Housing_needs_baseline_&year.;
+
+	 *adjust housing costs for inflation; 
+
+	  %dollar_convert( rentgrs, rentgrs_a, &year., 2017, series=CUUR0000SA0L2 )
+	  %dollar_convert( owncost, owncost_a, &year., 2017, series=CUUR0000SA0L2 )
+	  %dollar_convert( valueh, valueh_a, &year., 2017, series=CUUR0000SA0L2 )
+
+  	** Cost-burden flag & create cost ratio **;
+	    if ownershpd in (21, 22)  then do;
+
+			if hhincome_a > 0 then Costratio= (rentgrs_a*12)/hhincome_a;
+			  else if hhincome_a = 0 and rentgrs_a > 0 then costratio=1;
+			  else if hhincome_a =0 and rentgrs_a = 0 then costratio=0; 
+			  else if hhincome_a < 0 and rentgrs_a >= 0 then costratio=1; 
+			  			  
+		end;
+
+	    else if ownershpd in ( 12,13 ) then do;
+			if hhincome_a > 0 then Costratio= (owncost_a*12)/hhincome_a;
+			  else if hhincome_a = 0 and owncost_a > 0 then costratio=1;
+			  else if hhincome_a =0 and owncost_a = 0 then costratio=0; 
+			  else if hhincome_a < 0 and owncost_a >= 0 then costratio=1; 
+		end;
+	    
+			if Costratio >= 0.3 then costburden=1;
+		    else if HHIncome_a~=. then costburden=0;
+			if costratio >= 0.5 then severeburden=1;
+			else if HHIncome_a~=. then severeburden=0; 
+
+		tothh = 1;
+
+    
+    ****** Rental units ******;
+    
+   if ownershpd in (21, 22) then do;
+        
+    Tenure = 1;
+
+	 *create maximum desired or affordable rent based on HUD_Inc categories*; 
+    /* need to discuss for NC, use hudinc for now*/
```

Comment:
you need to compute the averages for NC for each hud_income group to determine % of income for hud_inc = 4, 5, and 6.